extensions/wikia/Wall/WallExternalController.class.php

```diff
@@ -277,6 +277,10 @@ public function deleteMessage() {
 			break;
 
 			case 'remove':
+				if (!$mw->canModerate($this->wg->User)) {
```

Comment:
I'm a little confused by this conditional (and those on L373 and L405).  Why is it that we need to call `load` on the WallMessage object only if the current user can't moderate?  Wouldn't this data be necessary to check any user?